### ISSUE TITLE
Missing ServiceInfo for webGateway causes startup failure.

### ISSUE THREAD

**Issue by ignasi35** [lagom/online-auction-java#17]
When installing the app in ConductR the `web-gateway` fails to start:

```

1) Could not find a suitable constructor in com.lightbend.lagom.javadsl.api.ServiceInfo. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at com.lightbend.lagom.javadsl.api.ServiceInfo.class(ServiceInfo.java:13)
  while locating com.lightbend.lagom.javadsl.api.ServiceInfo
    for parameter 2 at com.lightbend.lagom.internal.client.ServiceClientImplementor.<init>(ServiceClientImplementor.scala:44)
  while locating com.lightbend.lagom.internal.client.ServiceClientImplementor
    for parameter 3 at com.lightbend.lagom.internal.client.ServiceClientLoader.<init>(ServiceClientLoader.scala:16)
  while locating com.lightbend.lagom.internal.client.ServiceClientLoader
    for field at com.lightbend.lagom.internal.client.ServiceClientProvider.loader(ServiceClientLoader.scala:37)
  at com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport.bindClient(ServiceClientGuiceSupport.java:14) (via modules: com.google.inject.util.Modules$OverrideModule -> Module)

2) Could not find a suitable constructor in com.lightbend.lagom.javadsl.api.ServiceInfo. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at com.lightbend.lagom.javadsl.api.ServiceInfo.class(ServiceInfo.java:13)
  while locating com.lightbend.lagom.javadsl.api.ServiceInfo
    for parameter 2 at com.lightbend.lagom.internal.client.ServiceClientImplementor.<init>(ServiceClientImplementor.scala:44)
  while locating com.lightbend.lagom.internal.client.ServiceClientImplementor
    for parameter 3 at com.lightbend.lagom.internal.client.ServiceClientLoader.<init>(ServiceClientLoader.scala:16)
  while locating com.lightbend.lagom.internal.client.ServiceClientLoader
    for field at com.lightbend.lagom.internal.client.ServiceClientProvider.loader(ServiceClientLoader.scala:37)
  at com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport.bindClient(ServiceClientGuiceSupport.java:14) (via modules: com.google.inject.util.Modules$OverrideModule -> Module)

3) Could not find a suitable constructor in com.lightbend.lagom.javadsl.api.ServiceInfo. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at com.lightbend.lagom.javadsl.api.ServiceInfo.class(ServiceInfo.java:13)
  while locating com.lightbend.lagom.javadsl.api.ServiceInfo
    for parameter 2 at com.lightbend.lagom.internal.client.ServiceClientImplementor.<init>(ServiceClientImplementor.scala:44)
  while locating com.lightbend.lagom.internal.client.ServiceClientImplementor
    for parameter 3 at com.lightbend.lagom.internal.client.ServiceClientLoader.<init>(ServiceClientLoader.scala:16)
  while locating com.lightbend.lagom.internal.client.ServiceClientLoader
    for field at com.lightbend.lagom.internal.client.ServiceClientProvider.loader(ServiceClientLoader.scala:37)
  at com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport.bindClient(ServiceClientGuiceSupport.java:14) (via modules: com.google.inject.util.Modules$OverrideModule -> Module)

4) Could not find a suitable constructor in com.lightbend.lagom.javadsl.api.ServiceInfo. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at com.lightbend.lagom.javadsl.api.ServiceInfo.class(ServiceInfo.java:13)
  while locating com.lightbend.lagom.javadsl.api.ServiceInfo
    for parameter 2 at com.lightbend.lagom.internal.client.ServiceClientImplementor.<init>(ServiceClientImplementor.scala:44)
  while locating com.lightbend.lagom.internal.client.ServiceClientImplementor
    for parameter 3 at com.lightbend.lagom.internal.client.ServiceClientLoader.<init>(ServiceClientLoader.scala:16)
  while locating com.lightbend.lagom.internal.client.ServiceClientLoader
    for field at com.lightbend.lagom.internal.client.ServiceClientProvider.loader(ServiceClientLoader.scala:37)
  at com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport.bindClient(ServiceClientGuiceSupport.java:14) (via modules: com.google.inject.util.Modules$OverrideModule -> Module)

4 errors
at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:466)
at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:155)
at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107)
at com.google.inject.Guice.createInjector(Guice.java:96)
at com.google.inject.Guice.createInjector(Guice.java:84)
at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:181)
at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:123)
at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
at com.typesafe.conductr.bundlelib.play.api.ConductRApplicationLoader.load(ConductRApplicationLoader.scala:21)
at play.core.server.ProdServerStart$.start(ProdServerStart.scala:47)
at play.core.server.ProdServerStart$.main(ProdServerStart.scala:22)
at play.core.server.ProdServerStart.main(ProdServerStart.scala)
17:14:08  accd4f3a7956  Component webgateway exited with 255
Stopping bundle
```

That's a play App trying to invoke `bindClient()` but there's no `ServiceInfoProvider` available causing the Guice creation to fail.

Creating a `ServiceInfoProvider` in `ServiceClientGuiceSupport` has some undesired effects on ServiceLocator and ServiceRegistry.

**Comment by ignasi35** [comment]
This is blocked by https://github.com/lagom/lagom/pull/349

**Comment by ignasi35** [comment]
With current PR merges this is almost ready.

We still need to setup proper `sbt-conductr` plugin and test.

**Comment by ignasi35** [comment]
This was fixed in a842cb375e58d64ad05e28876cd0a2fd7914dbd9. See https://github.com/lagom/online-auction-java/blob/master/web-gateway/app/Module.java#L13
